scene/main/viewport.h

```diff
@@ -252,7 +252,7 @@ class Viewport : public Node {
 	Transform2D stretch_transform;
 
 	Size2i size = Size2i(512, 512);
-	Size2i size_2d_override;
+	Size2 size_2d_override;
```

Comment:
Suggesting reversion. The MRP is fixed without `size_2d_override` being a `Size2`.

```suggestion
	Size2i size_2d_override;
```

Reply:
This would mean that the size_2d_override that we can get from the Viewport class, and the size_2d_override that we are actually using for that viewport does not match up anymore.

I already hate leaving the code in a state where this is not exposed to the users, but also not exposing it to engine internals just makes it worse in my opinion.

Reply:
That's not quite right. The `final_size_override` that is passed to `_set_size` is a result of math (division of size by content scale) rather than explicit intent. So, there is nothing to match to or expose in the first place. And, it's the only consideration for fixing the MRP.

So, `final_size_override` and the `_set_size` function that it's passed to are the only necessary changes. Anything else is beyond the scope of fixing the MRP and can be considered in a future PR.

Reply:
No, the reason it needs to be a float is because the original assumption that size_2d_override (aka the size of the viewport in render-space) can be described as a Vector2i just doesn't hold in general. A side effect of this is the bug that we are fixing in the first place.

By changing only that one part where the bug manifests leaves the class in an inconsistent state, where certain assumptions that you would expect won't be correct anymore.

For example, stretch_transform.scale != size/size_2d_override which is a very reasonable thing to expect, that other code might depend on.

Reply:
That's a good point. `_set_size` saves the passed in override to the `Viewport`'s `size_2d_override` (truncated, if only the parameter is `Size2`) after it's used to calculate the `stretch_transform` scale. So, there would be some data loss there that is out of sync with the new `stretch_transform`.